### PR TITLE
[ISSUE #570] ASoC runtime optimization: Cassandra connectors

### PR DESCRIPTION
**Pull request by imaffe**
## What is the purpose of the change

1. implement Cassandra source/sink connectors

## Brief changelog

1. added Cassandra sink and source connector implementations

## Verifying this change

XXXX

Follow this checklist to help us incorporate your contribution quickly and easily. Notice, `it would be helpful if you could finish the following 5 checklist(the last one is not necessary)before request the community to review your PR`.

- [x] Make sure there is a [Github issue](https://github.com/apache/rocketmq/issues) filed for the change (usually before you start working on it). Trivial changes like typos do not require a Github issue. Your pull request should address just this issue, without pulling in other changes - one PR resolves one issue. 
- [x] Format the pull request title like `[ISSUE #123] Fix UnknownException when host config not exist`. Each commit in the pull request should have a meaningful subject line and body.
- [x] Write a pull request description that is detailed enough to understand what the pull request does, how, and why.
- [x] Write necessary unit-test(over 80% coverage) to verify your logic correction, more mock a little better when cross module dependency exist. If the new feature or significant change is committed, please remember to add integration-test in [test module](https://github.com/apache/rocketmq/tree/master/test).
- [x] Run `mvn -B clean apache-rat:check findbugs:findbugs checkstyle:checkstyle` to make sure basic checks pass. Run `mvn clean install -DskipITs` to make sure unit-test pass. Run `mvn clean test-compile failsafe:integration-test`  to make sure integration-test pass.
- [ ] If this contribution is large, please file an [Apache Individual Contributor License Agreement](http://www.apache.org/licenses/#clas).

### CODE REVIEW

**Comment by imaffe** [comment]
1. the rat check failed because of a `requirment.txt`, but I looked at other Apache python projects (like airflow), it seems one file like this without license declaration is ok

File: rocketmq-connect-cassandra/pom.xml [hunk 1]
```diff
@@ -0,0 +1,283 @@
...
+	OF ANY KIND, either express or implied. See the License for the specific
+	language governing permissions and limitations under the License. -->
+
+<project xmlns="http://maven.apache.org/POM/4.0.0"
+         xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+         xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
+    <modelVersion>4.0.0</modelVersion>
+    <groupId>org.apache.rocketmq</groupId>
+    <artifactId>rocketmq-connect-cassandra</artifactId>
+    <version>0.0.0-SNAPSHOT</version>
```
Comment (RongtongJin) [inline rocketmq-connect-cassandra/pom.xml:19]: 0.0.0-SNAPSHOT is a meaningless version.

File: rocketmq-connect-cassandra/pom.xml [hunk 2]
```diff
@@ -0,0 +1,283 @@
...
+
+    <name>connect-cassandra</name>
+<!--    <url>https://github.com/apache/incubator-rocketmq-externals/tree/master/rocketmq-connect-jdbc</url>-->
```
Comment (RongtongJin) [inline rocketmq-connect-cassandra/pom.xml:22]: Delete useless comments. Or replace the link with the current project.

File: rocketmq-connect-cassandra/pom.xml [hunk 3]
```diff
@@ -0,0 +1,283 @@
...
+                <groupId>org.codehaus.mojo</groupId>
+                <artifactId>findbugs-maven-plugin</artifactId>
+                <version>3.0.4</version>
+            </plugin>
+            <plugin>
+                <artifactId>maven-assembly-plugin</artifactId>
+                <version>3.0.0</version>
+                <configuration>
+                    <archive>
+<!--                        TODO, need to change this main class here to cassandra-->
```
Comment (RongtongJin) [inline rocketmq-connect-cassandra/pom.xml:154]: Delete useless comments

File: rocketmq-connect-cassandra/pom.xml [hunk 4]
```diff
@@ -0,0 +1,283 @@
...
+        <dependency>
+            <groupId>com.datastax.oss</groupId>
+            <artifactId>java-driver-query-builder</artifactId>
+            <version>4.5.1</version>
+        </dependency>
+<!--        <dependency>-->
+<!--            <groupId>com.datastax.oss</groupId>-->
+<!--            <artifactId>java-driver-mapper-runtime</artifactId>-->
+<!--            <version>4.5.1</version>-->
+<!--        </dependency>-->
```
Comment (RongtongJin) [inline rocketmq-connect-cassandra/pom.xml:280]: Delete useless comments

**Pull request merged**